eng/scripts/Language-Settings.ps1

```diff
@@ -0,0 +1,116 @@
+$Language = "go"
+$packagePattern = "go.mod"
+# rewrite from ChangeLog-Operations.ps1 used in Get-ChangeLogEntriesFromContent for go uses vx.x.x as version number
+$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+\s+(?<version>v$([AzureEngSemanticVersion]::SEMVER_REGEX))(\s+(?<releaseStatus>\(.+\))))"
+# constants for go version fetch
+$GO_VERSION_REGEX = ".+\s*=\s*`".*v?(?<version>$([AzureEngSemanticVersion]::SEMVER_REGEX))`""
+$VERSION_FILE_SUFFIXS = "*constants.go", "*version.go"
+
+# rewrite from artifact-metadata-parsing.ps1 used in RetrievePackages for fetch go single module info
+function Get-go-PackageInfoFromPackageFile ($pkg, $workingDirectory)
+{
+    $workFolder = $pkg.Directory
+    $releaseNotes = ""
+    $namespaceName = $workFolder.Name
+    $rpName = $workFolder.Parent.Name
```

Comment:
I think this will not work for some packages like azcore or azidentity which doesn't have a service folder currently

Reply:
You might want to consider a regex similar to https://github.com/Azure/azure-sdk/blob/main/eng/scripts/Query-Azure-Packages.ps1#L148.